src/expr/src/vector_op/string_to_array.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use itertools::Itertools;
+use risingwave_common::array::ListValue;
+use risingwave_common::types::ScalarImpl;
+use risingwave_expr_macro::function;
+
+fn string_to_array_inner(s: &str, sep: Option<&str>) -> Vec<String> {
```

Comment:
We can return an iterator rather than vec because both callers would continue processing its elements.